packages/astro/src/i18n/index.ts

```diff
@@ -47,6 +78,21 @@ export function getLocalesBaseUrl({ base, locales, trailingSlash, format }: GetL
 	});
 }
 
+export function getLocaleAbsoluteUrlList({ site, ...rest }: GetLocaleAbsoluteUrl) {
+	const locales = getLocaleRelativeUrlList(rest);
+	return locales.map((locale) => {
+		if (site) {
+			if (site.endsWith('/') || locale.startsWith('/')) {
+				return `${site}${locale}`;
+			} else {
+				return `${site}/${locale}`;
+			}
```

Comment:
We have `@astrojs/internal-helpers/path` for operations like this. Not sure this is exactly what you need, but I'd reccommend using utils from this package or adding the util you need. https://github.com/withastro/astro/blob/main/packages/internal-helpers/src/path.ts#L63-L76